ompi/mca/part/persist/part_persist.h

```diff
@@ -234,16 +234,16 @@ mca_part_persist_progress(void)
 
             if(done) {
                 size_t dt_size_;
-                int32_t dt_size;
+                uint32_t dt_size;
 
                 if(MCA_PART_PERSIST_REQUEST_PSEND == req->req_type) {
                     /* parse message */
                     req->world_peer  = req->setup_info[1].world_rank; 
 
                     err = opal_datatype_type_size(&(req->req_datatype->super), &dt_size_);
                     if(OMPI_SUCCESS != err) return OMPI_ERROR;
-                    dt_size = (dt_size_ > (size_t) INT_MAX) ? MPI_UNDEFINED : (int) dt_size_;
-                    int32_t bytes = req->real_count * dt_size;
+                    dt_size = (dt_size_ > (size_t) UINT_MAX) ? MPI_UNDEFINED : (uint32_t) dt_size_;
```

Comment:
I don't quite understand this part. Since `dt_size` is `uint32_t` type, it cannot exceed `UINT_MAX`. Should this be `size_t dt_size; ...; dt_size > INT_MAX`?

Reply:
@jsquyres Whenever you get time, can you double check this piece?

Reply:
Sorry I misread the variable name. No issues.